### PR TITLE
Issue/35

### PR DESCRIPTION
**Pull request by hazmat345**
This fixes beer-garden/beer-garden#35

### CODE REVIEW

**Comment by hazmat345** [comment]
Marking this as WIP until I fix the nested role cycle problem for real. I suspected it wouldn't work but it turns out it's catastrophic.

**Comment by hazmat345** [comment]
This is getting close. There are a few changes from the role admin page that need to be applied to the user page as well (relating to how nested roles are updated / displayed).

Then we just need some sort of banner to display the response from the server.

File: brew_view/controllers/roles_api.py [hunk 1]
```diff
@@ -214,3 +228,26 @@ def post(self):
...
+
+    Returns:
+        None
+
+    Raises:
+        ModelValidationError: A cycle was detected
+    """
+    for role in new_role.roles:
+        if role == base_role:
+            raise ModelValidationError('NOT GOOD')
```
Comment (loganasherjones) [inline brew_view/controllers/roles_api.py]: Probably a better error message here.

Comment (hazmat345) [reply brew_view/controllers/roles_api.py]: yeah...

**Comment by hazmat345** [comment]
@loganasherjones :skull:

**Comment by loganasherjones** [comment]
API Changes look good to me

**Pull request merged**